drivers/w1/w1_shell.c

```diff
@@ -0,0 +1,413 @@
+/*
+ * Copyright (c) 2018 Diego Sueiro
+ * Copyright (c) 2022 Thomas Stranger
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <stdlib.h>
+
+#include <zephyr/drivers/w1.h>
+#include <zephyr/shell/shell.h>
+#include <zephyr/kernel.h>
+
+#define W1_SHELL_HELP                                                          \
+	"1-Wire commands\n"                                                    \
+	"  options:\n"                                                         \
+	"    -r Perform bus reset before executing cmd"
+
+#define W1DEV_X_NOT_FOUND "1-Wire device not found: \"%s\""
+
+#define BUF_SIZE CONFIG_W1_SHELL_BUFFER_SIZE
+static uint8_t msg_buf[BUF_SIZE];
+
+static const char *w1_settings_name[W1_SETINGS_TYPE_COUNT] = {
+	[W1_SETTING_SPEED] = "speed",
+	[W1_SETTING_STRONG_PULLUP] = "spu",
+};
+
+static int read_io_options(const struct shell *sh, int pos, char **argv, bool *reset)
+{
+	char *arg = argv[pos];
+
+	if (arg[0] != '-') {
+		return pos;
+	}
+
+	for (arg = &arg[1]; *arg; arg++) {
+		switch (*arg) {
+		case 'r':
+			*reset = true;
+			break;
+		default:
+			shell_error(sh, "Unknown option %c", *arg);
+			return -EINVAL;
+		}
+	}
+
+	return ++pos;
+}
+
+/* 1-Wire reset bus <device> */
+static int cmd_w1_reset_bus(const struct shell *sh, size_t argc, char **argv)
+{
+	const struct device *dev;
+	int ret;
+
+	dev = device_get_binding(argv[1]);
+	if (!dev) {
```

Comment:
is additional `device_is_ready` check necessary?

Reply:
Compared to `DEVICE_DT_GET`, `device_get_binding` only returns a pointer if the device was successfully initialized.